xhttp/client.go

```diff
@@ -55,13 +55,26 @@ func RetrierWithRequestTimeout(timeout time.Duration) RetrierOption {
 	}
 }
 
+// RetrierWithStatus will configure the retrier to retry when this specific status code is received.
+// Multiple of this option can be passed to configure multiple different status code.
+func RetrierWithStatus(status int) RetrierOption {
```

Comment:
This only lets us add status codes (not remove them) so it'll always retry on the default ones, right? Maybe it would be good have a way to override the defaults... we could define the method like this so you can replace all the codes:

    func RetrierWithStatus(statuses map[int]struct{}) RetrierOption

Reply:
I would just change it to be a slice of status in that case (no need to expose the internal map/set structure that is a little odd because Go does not have sets :laughing: ). I was wondering if we would ever really do that, the current default status codes are quite limited to things that usually we really want to retry (500 and 503). Do you already have a use case in mind ? or else we could use it for a while and then revisit this.

Reply:
Maybe we could add later a "RetrierWithStatuses" if necessary (that then is a set that replaces the internal set). If we only have the option that replaces everything then every time I need a new status I need to remember all the basic status code that I need to retry (like 503) and if I forget them it is broken x_x.

Reply:
Yeah, I see the value of having sensible defaults... The one status I'm wondering about is `http.StatusTooManyRequests`. We might want to handle that at a higher level.

Reply:
> Yeah, I see the value of having sensible defaults... The one status I'm wondering about is `http.StatusTooManyRequests`. We might want to handle that at a higher level.

hmm I see, I do get this status a lot from LLM services and in general I just retry after some time exactly as I do with Service unavailable (both can have a header Retry-After if Im not wrong, which could be used here too, to make it more precise maybe ?)...knowing where you are in a window of status too many requests can be tricky, most APIs don't provide precise information..so your only option is to keep retrying anyway or just failing.

What I mean is, what would be the logic you would do differently ? (besides handling a Retry-After header ?) If we already have this use case..an alternative would be to remove "http.StatusTooManyRequests" from default handling, if it is that specific.

Reply:
So far we don't have any special handling for 429 status in the new connectors. I think we do in some of the connectors in feedback-river; not sure how we're handling it there.

Reply:
Yeah...so far Im designing things around my own experience but Im aware it doesn't cover all use cases...but would be good to have a clear use case in mind...WDYT ? we remove it as a default status code to be on the safe side ? I don't mind defining it manually on my services.

Reply:
Yup, that's a good solution for now.